### PR TITLE
Add platform libraries DfciDeviceIdSupportLib and DfciUiSupportLib

### LINKED ISSUE #470: [Feature]: Support for DFCI In Q35
### Feature Overview

Dfci is missing a couple of platform libraries needed to support Dfci in the Qemu environment.

### Solution Overview

Add the missing libraries, and configure Qemu to enable Dfci

### Alternatives Considered

_No response_

### Urgency

Medium

### Are you going to implement the feature request?

I will implement the feature

### Do you need maintainer feedback?

No maintainer feedback needed

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mikeytdisco**
## Description

Fixes #470 

Dfci requires the platform to supply a couple of libraries.  The following libraries are added to Qemu package:

DfciDeviceIdSupportLib
DfciUiSupportLib

and replacing the Null libraries used in the QemuQ35Pkg.dsc file.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
  - Yes, it impacts functionality by enabling Dfci to work in Qemu.
- [ ] Impacts security?

- [ ] Breaking change?

- [ ] Includes tests?

- [ ] Includes documentation?

## How This Was Tested

Tested in a QemuQ35Pkg test system.

## Integration Instructions

N/A

### CODE REVIEW

File: QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+## @file
+# DfciDeviceIdSupportLib.inf
+#
+# NULL Library for DfciDeviceIdSupportLib.
```
Comment (os-d) [inline QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf:4]: If this is the Null lib, should we call it DfciDeviceIdSupportLibNull? Or is this comment left over from copying the null instance?

File: QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  DebugLib
+  BaseLib
+  MemoryAllocationLib
+  UefiBootServicesTableLib
+
+[Protocols]
+  gEfiSmbiosProtocolGuid
+
+[Depex]
+  TRUE
```
Comment (os-d) [inline QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf:44]: nit: newline at end of file

Should `gEfiSmbiosProtocolGuid` be in the Depex? We fail out of the function if we don't find it (and aren't initialized).

File: QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.c [hunk 3]
```diff
@@ -0,0 +1,199 @@
+/** @file
+DfciUiSupportLibNull.c
+
+NULL instance of the UiSupportLib.
```
Comment (os-d) [inline QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.c:4]: If this is the null instance, should it be called DfciUiSupportLibNull? Or, is this comment and line 2 copied from the null instance?

File: QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf [hunk 4]
```diff
@@ -0,0 +1,51 @@
+## @file
+# DfciUiSupportLib.inf
+#
+# This library contains NULL routines to support UI operations associated with DFCI.
```
Comment (os-d) [inline QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf:4]: Same question on null lib.

File: QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+#
+# NULL Library for DfciDeviceIdSupportLib.
+#
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+##
+
+[Defines]
+  INF_VERSION                    = 0x00010005
```
Comment (os-d) [inline QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf:12]: nit: For a new lib, should we use a newer `INF_VERSION`?

File: QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+[Protocols]
+  gMsSWMProtocolGuid
+
+
+[Guids]
+  gEfiEventPreReadyToBootGuid
+  gDfciStartOfBdsNotifyGuid
+
+[Depex]
+  TRUE
```
Comment (os-d) [inline QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf:51]: nit: newline at end of file

**Review by os-d — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

File: QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf (hunk 2 above)
Comment (mikeytdisco) [reply QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf:44]: No.  Dfci is event driven.  The functions that call for the Device Id occurr  at Ready To Boot.  Putting a depex on that protocol will disturb other things.  SMBIOS should be published before Ready to Boot.

File: QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf (hunk 1 above)
Comment (mikeytdisco) [reply QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf]: Fixed

File: QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.c (hunk 3 above)
Comment (mikeytdisco) [reply QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.c]: Not the NULL.

Comment (mikeytdisco) [reply QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.c]: And fixed

File: QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf (hunk 4 above)
Comment (mikeytdisco) [reply QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf]: Fixed

File: QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf (hunk 5 above)
Comment (mikeytdisco) [reply QemuPkg/Library/DfciDeviceIdSupportLib/DfciDeviceIdSupportLib.inf]: Fixed

File: QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf (hunk 6 above)
Comment (mikeytdisco) [reply QemuPkg/Library/DfciUiSupportLib/DfciUiSupportLib.inf]: Fixed

**Pull request merged**